browser/net/brave_static_redirect_network_delegate_helper.cc

```diff
@@ -72,6 +100,16 @@ int OnBeforeURLRequest_StaticRedirectWork(
   if (!is_url_allowed) {
     LOG(ERROR) << "URL not allowed from system network delegate: " << gurl;
   }
+
+  bool is_url_blacklisted = std::any_of(blacklisted_patterns.begin(), blacklisted_patterns.end(),
+    [&gurl](URLPattern pattern) {
+      if (pattern.MatchesURL(gurl)) {
+        return true;
+      }
+      return false;
+    });
+  DCHECK(!is_url_blacklisted) << gurl.spec();
+
```

Comment:
I don't think we should enforce a blacklist here. If the URL remapping logic is solid (and backed by tests, thank you), then this check is unnecessary anyway.